### ISSUE TITLE
`uc_mem_unmap` fails to unmap the start of a region if it is directly preceeded by another region

### ISSUE THREAD

**Issue by Kritzefitz** [unicorn-engine/unicorn#1495]
This test case fails unexpectedly, when I run it on 907ec5095d10eb4488547c1d9db08230f73f377e:

```C
static void test_x86_splitting_unmap()
{
  uc_engine *uc;

  OK(uc_open(UC_ARCH_X86, UC_MODE_32, &uc));

  OK(uc_mem_map(uc, 0x20000, 0x1000, UC_PROT_NONE));
  OK(uc_mem_map(uc, 0x21000, 0x2000, UC_PROT_NONE));

  OK(uc_mem_unmap(uc, 0x21000, 0x1000));

  OK(uc_close(uc));
}
```

The error message is:

```
Test test_x86_mmio_unmap...                     [ FAILED ]
  test_x86.c:589: Check __err == UC_ERR_OK... failed
    No memory available or memory not present (UC_ERR_NOMEM)
```

The test case does not fail if the first call to `uc_mem_map` ( `OK(uc_mem_map(uc, 0x20000, 0x1000, UC_PROT_NONE));`) is removed.

I would expect that unmapping parts of a region doesn't depend on the existence of other regions in the vicinity. So I think this is a bug.

**Comment by wtdcode** [comment]
Looks like a bug indeed, could you create a PR to add your tests case to the corresponding file?

**Comment by wtdcode** [comment]
Fixed in c733bbada356b0373fa8aa72c044574bb855fd24

**Comment by Kritzefitz** [comment]
Thank you for fixing this so quickly!
